### PR TITLE
Fixes #30748 - On Ubuntu 18.04+ use /etc/netplan

### PR DESCRIPTION
**Pull request by stefanlasiewski**

### CODE REVIEW

**Comment by stefanlasiewski** [comment]
Previous PR was https://github.com/theforeman/foreman/pull/7947, but I had a merge issue and will simply resubmit.

File: app/views/unattended/provisioning_templates/snippet/preseed_networking_setup.erb [hunk 1]
```diff
@@ -70,4 +75,25 @@ iface $real inet6 static
...
 EOF
+<% else -%>
+<%# Ubuntu 18.04+ uses /etc/netplan %>
+cat << EOF >> /etc/netplan/01-netcfg.yaml
+    # <%= interface.identifier %>
+    $real:
+  <% if interface_dhcp -%>
+        dhcp4: true
+  <% else -%>
+      addresses:
```
Comment (lzap) [inline app/views/unattended/provisioning_templates/snippet/preseed_networking_setup.erb:85]: Identation seems to be a bit off, I need to test this.

**Comment by lzap** [comment]
ok to test

**Comment by lzap** [comment]
If you run 

    RAILS_ENV=test bundle exec rake snapshots:generate

you will see how this change modified the output. Please remove the extra whitespace, it's unnecessary.

File: app/views/unattended/provisioning_templates/snippet/preseed_networking_setup.erb [hunk 2]
```diff
@@ -13,6 +13,9 @@ description: this will configure your host networking, it configures your primar
 <% host_dhcp6 = host_subnet6.nil? ? true : host_subnet6.dhcp_boot_mode? -%>
 
 real=`ip -o link | awk '/<%= @host.mac -%>/ {print $2;}' | sed s/://`
+
+<%# Ubuntu before 18.04 uses /etc/network/interfaces %>
```
Comment (lzap) [inline app/views/unattended/provisioning_templates/snippet/preseed_networking_setup.erb]: This line is causing the extra whitespace:

```suggestion
<%# Ubuntu before 18.04 uses /etc/network/interfaces -%>
```

File: app/views/unattended/provisioning_templates/snippet/preseed_networking_setup.erb [hunk 3]
```diff
@@ -70,4 +75,25 @@ iface $real inet6 static
 <% end -%>
 <% end -%>
 EOF
+<% else -%>
+<%# Ubuntu 18.04+ uses /etc/netplan %>
```
Comment (lzap) [inline app/views/unattended/provisioning_templates/snippet/preseed_networking_setup.erb]: Here as well I think.

Comment (stefanlasiewski) [reply app/views/unattended/provisioning_templates/snippet/preseed_networking_setup.erb]: Thanks for the tips! I added these.

**Comment by lzap** [comment]
Tests are still failing, have you executed the snapshot regeneration? It looks like we've missed one extra empty line, can you get rid of it?

```
 real=`ip -o link | awk '/00-f0-54-1a-7e-e0/ {print $2;}' | sed s/://`
+
 cat << EOF > /etc/network/interfaces
 #loopback
 auto lo
Diff for /home/jenkins/workspace/test_develop_pr_core/database/postgresql/ruby/2.7/slave/fast/test/unit/foreman/renderer/snapshots/ProvisioningTemplate/snippet/preseed_networking_setup.snap.txt:
--- expected
+++ actual
@@ -1,6 +1,7 @@
 "
 
 real=`ip -o link | awk '/00-f0-54-1a-7e-e0/ {print $2;}' | sed s/://`
+
 cat << EOF > /etc/network/interfaces
 #loopback
 auto lo
```

**Comment by stefanlasiewski** [comment]
Unfortunately, I can't get the Foreman development environment to work on my Mac due to a lack of libvirt postgresql support, so I had to do it the old fashioned way.

Let me try and track it down and add the fix.

**Comment by stefanlasiewski** [comment]
It's running clean for me now. I can't tell why the `upgrade` & `katello` tests failed above, and I'm not sure how to retrigger the test.

The `Legacy JS` tag may have been added when I made a mistake during a rebase. (I'm still learning that part of github workflows)

```
stefanl@stefanl:foreman $ git branch --show-current
30748_ubuntu-add-netplan_2
stefanl@stefanl:foreman $ RAILS_ENV=test bundle exec rake snapshots:generate
The Apipie cache is turned off. Enable it and run apipie:cache rake task to speed up API calls.
stefanl@stefanl:foreman $
```

**Comment by lzap** [comment]
Sorry for the delay. I was wrong, it's not necessary to create snapshots for this case. Just rebase on develop to get tests passing, it failed for a different reason.

**Comment by stefanlasiewski** [comment]
@lzap I rebased & pushed.  Pretty sure I got the rebase correct after a couple bad tries.

I can't tell why `Assert Single Commit (non-blocking) / build (pull_request)` is getting triggered.

**Review by lzap — APPROVED** [review]
Sorry for the delays, thanks.

**Comment by tbrisker** [comment]
2.4 - 1a4262503

File: app/views/unattended/provisioning_templates/snippet/preseed_networking_setup.erb [hunk 4]
```diff
@@ -70,4 +74,25 @@ iface $real inet6 static
...
+<% else -%>
+<%# Ubuntu 18.04+ uses /etc/netplan -%>
+cat << EOF >> /etc/netplan/01-netcfg.yaml
+    # <%= interface.identifier %>
+    $real:
+  <% if interface_dhcp -%>
+        dhcp4: true
+  <% else -%>
+      addresses:
+      <% addresses.each do |address| -%>
```
Comment (lzap) [inline app/views/unattended/provisioning_templates/snippet/preseed_networking_setup.erb:86]: Addresses is undefined, which I missed during my review :-(

**Comment by lzap** [comment]
I am proposing to revert this patch, it is not working correctly: https://github.com/theforeman/foreman/pull/8658

**Pull request merged**